pkg/haprovider/haprovider.go

```diff
@@ -138,21 +138,40 @@ func (r *HAProvider) annotateService(ctx context.Context, cluster *clusterv1.Clu
 		return serviceAnnotation, nil
 	}
 
-	// TODO(iXinqi): Temporarily commenting adding annotation to HA svc, will uncomment it after the feature is fully tested.
-	//aviInfraSetting, err := r.getAviInfraSettingFromCluster(ctx, cluster)
-	//if err != nil {
-	//	return serviceAnnotation, err
-	//}
-
-	//if aviInfraSetting != nil {
-	//	// add AVIInfraSetting annotation when creating HA svc
-	//	serviceAnnotation[akoov1alpha1.HAAVIInfraSettingAnnotationsKey] = aviInfraSetting.Name
-	//}
+	//check if mgmt cluster and then if there should a ais created from adc, then wait
+
+	adcForCluster, err := r.getADCForCluster(ctx,cluster)
+	if err != nil {
+		return serviceAnnotation, err
+	}
+	//no adc is selected for cluster, no annotation is needed.
+	if adcForCluster == nil {
+		return serviceAnnotation, nil
+	}
+
+	aviInfraSetting, err := r.getAviInfraSettingFromAdc(ctx, adcForCluster)
+	if err != nil {
+		return serviceAnnotation, err
+	}
+
+	if _, ok := cluster.Labels[akoov1alpha1.ManagememtClusterRoleLabel]; ok {
+		// this is mgmt cluster
+
+		if adcForCluster.Spec.ControlPlaneNetwork.Name != "" && adcForCluster.Spec.ControlPlaneNetwork.Name != adcForCluster.Spec.DataNetwork.Name {
```

Comment:
I'd prefer using cidr instead of name, cuz I know cidr is definitely unique, not very sure about name